### PR TITLE
minor document related improvements to make it easier to consume

### PR DESCRIPTION
**Pull request by mswiderski**

### CODE REVIEW

File: kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/model/instance/DocumentInstanceList.java [hunk 1]
```diff
@@ -29,7 +29,7 @@
 @XmlRootElement(name = "document-instance-list")
 public class DocumentInstanceList implements ItemList<DocumentInstance> {
 
-    @XmlElement(name="document-instance")
+    @XmlElement(name="document-instances")
```
Comment (psiroky) [inline kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/model/instance/DocumentInstanceList.java]: Why the change to plural?

Comment (mswiderski) [inline kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/model/instance/DocumentInstanceList.java]: because it represents a list/array so it better matches the wrapper when marshalled

Comment (psiroky) [inline kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/model/instance/DocumentInstanceList.java]: Hmm, I thought that the @XmlElement defines the name for individual XML tags (not the wrapping collection tag). What wrapper do you mean? (I ma likely just not seeing something, but would like to understand what is going on :))

Comment (mswiderski) [inline kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/model/instance/DocumentInstanceList.java]: it is used by both jaxb and json

File: kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/DocumentServiceBase.java [hunk 2]
```diff
@@ -117,13 +117,13 @@ public DocumentInstanceList listDocuments(Integer page, Integer pageSize) {
...
         for (Document doc : documents) {
-            list.add(convertDocument(doc));
+            list.add(convertDocument(doc, false));
         }
 
         return list;
     }
 
-    protected DocumentInstance convertDocument(Document document) {
+    protected DocumentInstance convertDocument(Document document, boolean withContent) {
```
Comment (psiroky) [inline kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/DocumentServiceBase.java]: Tbh, I am not a big fan of these boolean flagging args :) I would prefer enum (even with just two values). But I won't hold the PR for this, since it is largely matter of personal style.

**Comment by psiroky** [comment]
Looks good. +1
